### PR TITLE
[SHIRO-685] Potential NullPointerException if PermissionResolver retu…

### PR DESCRIPTION
**Pull request by fpapon**
…rn null/empty string.

Made changes according to the JIRA and I added a test case.

@bdemers can you review please?

### CODE REVIEW

File: core/src/test/java/org/apache/shiro/realm/AuthorizingRealmTest.java [hunk 1]
```diff
@@ -214,6 +228,36 @@ public void testRealmWithRolePermissionResolver()
...
+        realm.setRolePermissionResolver( new RolePermissionResolver()
+        {
+            public Collection<Permission> resolvePermissionsInRole( String roleString )
+            {
+                Collection<Permission> permissions = new HashSet<Permission>();
+                if( roleString.equals( ROLE ))
+                {
+                    permissions.add( new WildcardPermission( ROLE + ":perm1" ) );
+                    permissions.add( new WildcardPermission( ROLE + ":perm2" ) );
+                    permissions.add( new WildcardPermission( "other:*:foo" ) );
```
Comment (bdemers) [inline core/src/test/java/org/apache/shiro/realm/AuthorizingRealmTest.java]: Looks good!

Couple minor nits:
We should probably add a null/empty string here too?
and maybe a whitespace string to both? `" "` `"\t "`? (that might be covered in existing tests though)

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Shiro-pr/75/

File: core/src/test/java/org/apache/shiro/realm/AuthorizingRealmTest.java (hunk 1 above)
Comment (fpapon) [reply core/src/test/java/org/apache/shiro/realm/AuthorizingRealmTest.java]: @bdemers done :)

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Shiro-pr/76/

**Pull request merged**